src/System.Runtime.Extensions/src/System/IO/Path.cs

```diff
@@ -189,8 +168,9 @@ public static string GetFileNameWithoutExtension(string path)
         public static string GetPathRoot(string path)
         {
             if (path == null) return null;
-            path = NormalizePath(path, fullCheck: false);
-            return path.Substring(0, PathInternal.GetRootLength(path));
+            PathInternal.CheckInvalidPathChars(path);
+            int pathRoot = PathInternal.GetRootLength(path);
+            return pathRoot <= 0 ? String.Empty : path.Substring(0, pathRoot);
```

Comment:
We should probably move GetPathRoot into the PAL layer.  On Unix, GetPathRoot is going to either return the empty string for relative paths or "/" for absolute paths, which means we should never need to be allocating any new strings.  If you'd prefer this be done as a separate change, that's fine... just open an issue to me to fix it.